### ISSUE TITLE
[non-mixer] Idea: knobs option in the "control" strip

### ISSUE THREAD

**Issue by pevsner** [original-male/non#189]
It would be great to be able to have the option of using "knob" widgets in the "control" strip. Making best use of screen real estate, then it'd be easy to set up a nice usable channel strip accessing plugins.
It would be super cool if there was and option to "enable/disable" preferably a whole plugin but probably easiest individual controls. It's so much easier tweaking in a strip than popping out plugins all the time. I basically use only an eq and a compressor ( and saturation) on most instruments. I did a test recently with the parametric eq plugin/fil4, sc4 comp and tap tubewarmth in strips and was able to get a rockin' mix in no time.It's just the sliders use us so much real estate. 2 or 3 knobs in a row would work well. 
![non-channelstrip](https://cloud.githubusercontent.com/assets/553646/13654567/0b47d54a-e651-11e5-805a-d5a5bd9a790f.jpg)
.

**Comment by pevsner** [comment]
I see in the mixbus manual there tone control is:
HI: 2000 Hz Q=1.0 Shelf
MID: 800 Hz Q=0.7 Bell
LO: 300 Hz Q=1.0 Shelf
An integrated widget for the "control " strip calling upon the parametric eq plugin with those settings preloaded would be cool.

**Comment by original-male** [comment]
@pevsner It actually used to look pretty much just like this. However, your mockup illustrates the problem: there's very little room for labels.

**Comment by pevsner** [comment]
@original-male Yes I imagine, no way of abbreviating the labels to idk just the first letter? Would you mind slipping the knobs back as an option? its so much simpler being able to see and tweak the whole lot old school mixing desk style. Also the default gain has "bind" ticked but doesn't show as a control. I have to add another gain module to tweak in the strip. Pan seems fine(mind you not default).
Off subject.. the ambisonics you have integrated rocks... is there any chance you could maybe have a variable snap to a grid on the spatializer, maybe just activated by [shift] key,-or snap to "0" when close to it (z axis) for (bass snare kick) also the instrument points seem to leap about when you first move them. Cheers. I use non as it's the cleanest sounding daw.

**Comment by pevsner** [comment]
@jrigg  I agree about the flexibility. I was basically just requesting the knob widget to be an option in the control strip really, possibly with the ability to switch on/off. The widget thing was just an idea and only available if that plugin was actually present in the "chain". It was just really a real estate suggestion as you can get a couple/three of knobs in a row, making using it as a channel strip easier. As male points out there's a space issue with labels. A possible solution to for that could be an option (in the plugin knob/vert/horiz menu) to abbreviate labels to say 4 letters, and also activated in the control strip. An abbreviate option might be also good generally.

**Comment by pevsner** [comment]
I might try to implement this myself, although my C skills are extremely limited. Just having the option of knobs would be good to have. 
![non-channelstrip1](https://cloud.githubusercontent.com/assets/553646/13714351/f72dc940-e7c5-11e5-90cc-5167fa2fc251.png)

**Comment by original-male** [comment]
Out of curiosity, why do you think knobs would be better than sliders? If you go back in time to older versions of non-mixer, it did use knobs for the controls view. The sliders were meant to be an improvement.

**Comment by pevsner** [comment]
@original-male Just in terms of space. It would be good to have both options to be honest.
If I set up a simple channel strip using an eq (set up low/mid/hig with finer tweaks from the plugin that would be 3-6 sliders, but more to do a fuller eq setup)  and a compressor ( 6 sliders) and tube warmth 2 sliders, it becomes quite crowded
I feel a combination of knobs and sliders would be visually easier to handle. Eg A compressor could be all knobs except for make up gain (slider), which would add a visual separation to the next plugin.

**Comment by pevsner** [comment]
I'm not diss-ing the slider btw

**Comment by original-male** [comment]
Having a mix of sliders and knobs would be a lot more complicated (mainly because it would require a way to define which is which).

I actually have some (incomplete) custom plugin interface stuff developed to allow this kind of thing, but that would apply to the module parameter editor window, not to the strip controls view.

The problem with all of this from a design point of view is that it's a lot of work, for both developer (to build the interface) and the user (to customize it all) and it's hard to say if it would ever save any time in practice over just going into the module parameter editor/memorizing where all the controls are located.

**Comment by jrigg** [comment]
FWIW I consider the present behaviour close to perfect (not criticising pevsner's suggestion). I tend to use only horizontal sliders in the parameter editor and like to keep the vertical height of the mixer strips (in signal view; rarely use fader view) to a minimum. I arrange mixer channels in a single row along the bottom of the screen with the rest of the screen being used by non-timeline.

After using analogue mixing desks for a few decades before moving to computer based recording, I don't feel the need for a software mixer to look like a picture of a piece of hardware; in fact I find it offputting (same goes for plugins). What is ergonomic on a piece of hardware does not necessarily translate to a computer screen. I'm aware not everyone agrees, but the visual simplicity of the Non tools was a major factor in my decision to move to Non from a more conventional DAW.

**Comment by pevsner** [comment]
@original-male It was just a thought I aired. I had just been doing an insane amount of mixing and found that I could achieve a nice good balanced mix really quickly with just an eq and compressor on each strip. I thought that having both knobs and sliders would enable one to organise the strips better visually and more compactly. Not having to pop out plugins all the time kind of keeps your focus just on achieving a good overall balance I find.

**Comment by pevsner** [comment]
I'm fine with the gui as is, less is definitely more.
Being able to bind controls is a cool feature.
Another thought (below) which I would find very useful and then I'll stop.
![non-gain-pan](https://cloud.githubusercontent.com/assets/553646/13801185/70435700-eb27-11e5-965f-36b296c942d4.png)

**Comment by jrigg** [comment]
That would be useful :-)

BTW you could save a small amount of DSP load by putting the gain before the mono pan so the gain is only acting on the mono signal. That's how it's done in mono channels on an analogue mixer, despite the pan control usually being placed above the fader on the panel.

**Comment by pevsner** [comment]
@jrigg OK thanks. The reason I do it that way is sometimes I put an aux after the pan, before gain, (say for sending to a fx for both guitar, or stereo snare send), and thus the send gain remains constant. I'll do it on the other tracks in future.

**Comment by pevsner** [comment]
@original-male Actually, haha, one last thought, would it be at all possible to be able to remove the default meter on the chain strip. I quite like to mix without visual simulation sometimes, or just a vu/din needle meter/s.

**Comment by original-male** [comment]
@pevsner Well, I'm 50% sure you can edit the snapshot file and remove the lines that create the meter modules without breaking anything.

**Comment by original-male** [comment]
@pevsner As do your idea of having the module controls expanded in the chain view... I've had the same thought. It has a couple of problems: it's if you have a lot of modules or modules with many controls, then you'll be far less likely to be able to see them all at once, which leads into: the mousewheel is used to scroll the chain view and also to adjust the sliders, so it would be highly likely that you'd accidentally change parameters while attempting to scroll to see the entire chain. I couldn't come up with a good compromise (since I like and am very accustomed to both uses of the mousewheel).

Almost any solution seems like it would ruin the 'quick access' aspect of the idea.

**Comment by pevsner** [comment]
![non-gain-slider](https://cloud.githubusercontent.com/assets/553646/13826340/d0a3f2ee-ebae-11e5-86b0-f6d2010fe7c3.png)
My original thought was this actually, but thought being integrated  was less mouse clicks.
Click the button hot-spot  to expand/contract a slider, horizontal maybe for pan.
If that's not possible, how about -normal behaviour but - it opens a smaller than normal window without a titlebar (ie no decorations) that opens in the exact position of  "click" and closes upon "mouse out".?

**Comment by pevsner** [comment]
Clicking the hot spot/button locks the main chain in position until the slider is closed.
In fact you could do that on the integrated one with a secondary button on "gain". on/off locked main chain.

**Comment by pevsner** [comment]
@original-male Thank you, the meter removed seems stable, both chain and mixer ones go however, I was hoping to just remove the chain one, actually though it's quite useful in that I can just have a row of din jmeters ( I'm never going to peak as everythings about -14). A module able to "swollow" jmeters in the chain would be real cool (could manually hook it up - save a template).

**Comment by pevsner** [comment]
I switched from JWM to Fluxbox ( both horrid looking default, but can be themed well ), in which you can set windows to open "UnderMousePlacement" which is ideal for gain/pan and plugins in the signal strip. I rarely use the fader strip now.
Re-clicking a strip item doesn't seem to raise the plugin parameter popup as it does in JWM, but no biggie to click the toolbar(panel).
The tab grouping option in Fluxbox ( setup in "apps" config file - is then grouped when you start) makes for a nice clean uncluttered setup. Unfortunately with "NTL" you can't drag in audio when tabbed. No problem though, I record most tracks in straight, and just detach momentarily to do so when needed.
